Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs

```diff
@@ -517,10 +559,22 @@ public override void Update(float frameTime)
             base.Update(frameTime);
 
             var query = EntityQueryEnumerator<ActiveMicrowaveComponent, MicrowaveComponent>();
+
             while (query.MoveNext(out var uid, out var active, out var microwave))
             {
-                //check if there's still cook time left
+
                 active.CookTimeRemaining -= frameTime;
+
+                if (_gameTiming.CurTime >= _targetTime)
+                {
+                    if (!HandleUnsafeItems(uid, microwave))
+                    {
+                        _destruction.BreakEntity(uid);
+                    }
+                    _targetTime += TimeSpan.FromSeconds(1);
```

Comment:
Do you mean to add `continue` here to stop processing the rest of this loop?

Reply:
yeah, should it not be break though?

Reply:
Adding a `continue` is a pattern that is used elsewhere in the code base and would help soemone reading the code in the future. It's not obvious from one glance but the rest of the code after the time check is being run after every tick (instead of at one second intervals). I think it's important to be consistent with the rest of the code to avoid confusing others.

Reply:
continue seems to be causing test failures where as break does not.

Reply:
`break` is a bug, because that means breaking out of the while loop that iterates through all of the microwave components (i.e. exploding one is enough to stop processing the rest of the microwaves). You should investigate the root cause of the test failure, rather than saying "this code is correct because it passes the tests".

Reply:
Tests seem to be passing now, strange.

Reply:
```
if (metaData.EntityPrototype.ID == recipeSolid.Key)
                        {
                            _sharedContainer.Remove(item, component.Storage);
                            EntityManager.DeleteEntity(item);
                            continue;
                        }
```
Removed the break statement here and put a continue, seemed to resolve the test failure?

Reply:
nvm that wasn't it, lol.